src/traces/bar/cross_trace_calc.js

```diff
@@ -119,8 +123,35 @@ function setGroupPositions(gd, pa, sa, calcTraces, opts) {
             }
             break;
     }
+    collectExtents(calcTraces, pa, opts);
+}
+
+function standardizeCornerradius(calcTraces) {
+    if(!calcTraces.length) return;
+    var i, calcTrace, fullTrace;
+    var firstTrace = calcTraces[0][0].trace;
```

Comment:
Do we pick the correct first trace if `trace[0].visible` is set to *false*?

Reply:
Yes, verified — non-visible traces are excluded earlier on line 32

Reply:
Ok. This only matters if there are differences among the traces, so we should document that best practice is to use `layout.barcornerradius` when possible, and otherwise to set all traces in a given stack with matching values.